torch/_subclasses/fake_tensor.py

```diff
@@ -515,7 +519,10 @@ def nonzero(fake_mode, func, arg):
         # disjoint with what can actually occur.  But this is fine:
         # remember, the hypothesis is that if your later code works
         # with N >= 2, it will work with N = 1 and N = 0.
-        constrain_range(nnz, min=2, max=sys.maxsize - 1)
+        maxval = sys.maxsize - 1
+        if not free_symbols(arg.numel()):
+            maxval = arg.numel()
```

Comment:
Actually, technically we can do even better, by multiplying the max range of symbolic variables to compute an approximate max range here